### PR TITLE
Updated to support snapcraft 2.30.

### PR DESCRIPTION
**Pull request by flexiondotorg**
**Do not merge this until Snapcraft 2.30 is available for use via Launchpad**

This merge proposal modifies the Atom snap to take advantage of the improved support for creating classic snaps from pre-built binaries.

The Atom snap has been rebuilt and published to the snap store using snapcraft from the edge channel. But the Atom snapcrafters GitHub repository is connected to https://build.snapcraft.io which builds via Launchpad. The version of Snapcraft currently available in Launchpad is not sufficient to work with these modifications to the yaml and launcher.

### CODE REVIEW

File: monitor.sh [hunk 1]
```diff
@@ -37,6 +37,6 @@ fi
 if [ ${REBUILD} -eq 1 ]; then
     mv snap/snapcraft.yaml.new snap/snapcraft.yaml
     echo "OLD_VERSION=${VERSION}" > OLD_VERSION
-    git commit -m "Version bumped to ${VERSION}"
-    git push
+    #git commit -m "Version bumped to ${VERSION}"
+    #git push
```
Comment (evandandrea) [inline monitor.sh:41]: Pet peeve. Don't comment lines out in git, just delete them.

File: files/bin/electron-launch [hunk 2]
```diff
@@ -1,24 +1,4 @@
...
-            ;;
-    esac
-
-    export LD_LIBRARY_PATH=$SNAP/usr/lib:$SNAP/usr/lib/$TRIPLET:$LD_LIBRARY_PATH
-    export LD_LIBRARY_PATH=$SNAP/lib:$SNAP/lib/$TRIPLET:$LD_LIBRARY_PATH
-fi
-
-exec ${SNAP}/bin/desktop-launch $@
+[ -n "$XDG_RUNTIME_DIR" ] && mkdir -p $XDG_RUNTIME_DIR -m 700
+exec "$@" --executed-from="$(pwd)" --pid=$$ > /dev/null 2>&1 &
```
Comment (evandandrea) [inline files/bin/electron-launch:4]: Why do you background this process?

Comment (flexiondotorg) [reply files/bin/electron-launch:4]: To fix #2

**Comment by flexiondotorg** [comment]
This is in master now.

**Pull request closed without merging**